actors/builtin/miner/deadline_state.go

```diff
@@ -889,3 +883,120 @@ func (dl *Deadline) ProcessPoSt(store adt.Store, quant QuantSpec, faultExpiratio
 	dl.PostSubmissions = abi.NewBitField()
 	return newFaultyPower, failedRecoveryPower, nil
 }
+
+type PoStResult struct {
+	NewFaultyPower, RetractedRecoveryPower, RecoveredPower PowerPair
+	Sectors, IgnoredSectors                                *bitfield.BitField
+}
+
+// PowerDelta returns the power change (positive or negative) after processing
+// the PoSt submission.
+func (p *PoStResult) PowerDelta() PowerPair {
+	return p.RecoveredPower.Sub(p.NewFaultyPower)
+}
+
+// PenaltyPower is the power from this PoSt that should be penalized.
+func (p *PoStResult) PenaltyPower() PowerPair {
+	return p.NewFaultyPower.Add(p.RetractedRecoveryPower)
+}
+
+func (dl *Deadline) ProcessWindowedPoSt(
```

Comment:
Consider `ProcessSectorsToBeProven` or `ProcessProvenSectors` to clarify that this has nothing to do with proof validation.

Reply:
`RecordProvenSectors`?